OSXDemo/OCTConversationViewController.m

```diff
@@ -0,0 +1,264 @@
+
+    [self.manager.chats sendMessageToChat:chat
+                                     text:sender.stringValue
+                                     type:OCTToxMessageTypeNormal
+                                    error:nil];
+    sender.stringValue = @"";
+
+}
+
+#pragma mark - NSTableViewDelegate
+
+- (NSView *) tableView:(NSTableView *)tableView
+    viewForTableColumn:(NSTableColumn *)tableColumn
+                   row:(NSInteger)row
+{
+    NSTextField *field = [tableView
+                          makeViewWithIdentifier:kCellIdent
+                                           owner:self];
+
+    if (! field) {
+        field = [NSTextField new];
+        field.identifier = kCellIdent;
+    }
+
+    field.selectable = YES;
+    field.editable = NO;
+
+
+    NSIndexPath *path = [NSIndexPath indexPathForRow:row inSection:0];
+
+    if (tableView == self.chatsTableView) {
+
+        OCTChat *chat = [self.chatResultsController objectAtIndexPath:path];
+        OCTFriend *friend = [chat.friends firstObject];
+
+        field.stringValue = (friend.isConnected) ? [NSString stringWithFormat : @"%@ : Online", friend.nickname] : friend.nickname;
+
+    }
+    else {
+        OCTMessageAbstract *messageAbstract = [self.conversationResultsController objectAtIndexPath:path];
+        if (messageAbstract.messageText) {
+            if (messageAbstract.sender) {
+                field.stringValue = [NSString stringWithFormat:@"%@: %@", messageAbstract.sender.nickname, messageAbstract.messageText.text];
+            }
+            else {
+                field.stringValue = [NSString stringWithFormat:@"%@: %@", self.manager.user.userName, messageAbstract.messageText.text];
+            }
+        }
+    }
+
+    return field;
+}
+
+- (CGFloat)tableView:(NSTableView *)tableView heightOfRow:(NSInteger)row
+{
+    return 50.0;
+}
+
+#pragma mark - NSTableViewDataSource
+
+- (NSInteger)numberOfRowsInTableView:(NSTableView *)tableView
+{
+    if (tableView == self.chatsTableView) {
+        return [self.chatResultsController numberOfRowsForSectionIndex:0];
+    }
+    else {
+        return [self.conversationResultsController numberOfRowsForSectionIndex:0];
+    }
+
+    return 0;
+}
+
+#pragma mark -  RBQFetchedResultsControllerDelegate
+
+- (void)controllerWillChangeContent:(RBQFetchedResultsController *)controller
+{
+    NSTableView *tableView = (self.chatResultsController == controller) ? self.chatsTableView : self.conversationTableView;
+    [tableView beginUpdates];
+}
+
+- (void) controller:(RBQFetchedResultsController *)controller
+    didChangeObject:(RBQSafeRealmObject *)anObject
+        atIndexPath:(NSIndexPath *)indexPath
+      forChangeType:(RBQFetchedResultsChangeType)type
+       newIndexPath:(NSIndexPath *)newIndexPath
+{
+    NSTableView *tableView = (self.chatResultsController == controller) ? self.chatsTableView : self.conversationTableView;
+
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Well, "Thread 1" means main queue. :-)
You can click on stack trace to see what's actually happening there. 

<img width="1025" alt="screen shot 2015-12-24 at 01 02 05" src="https://cloud.githubusercontent.com/assets/1132979/11987118/fee6a604-a9d9-11e5-9006-89ac75c791f5.png">